[T13-2] SSENISUB

valensia0711 opened the pull request:
@IamRENCE @DanKhoo @abtyx @NgeowXiuQi

xhsien commented:
Some comments from me:

1. Update project page.
2. For features not yet implemented, add `coming in v2.0` in user guide.

xhsien reviewed (approved):
Remember to add relevant diagrams (activity/class/object/sequence) in your developer guide to apply what we've learnt in class!

DanKhoo commented:
Hi @chrisboo, we have added some relevant diagrams, 3.2, 3.3, 3.4, 3.8 & 3.9 of the developer guide :)

abtyx committed:
Merge branch 'master' into v1.2privacyCommand

abtyx committed:
remove unused import statements

abtyx committed:
edit PrivacyCommand to take in the new fields Feedback, Salary etc

abtyx committed:
edit Phone equals method to check privacy
edit XmlAdaptedPerson to account for phone privacy
edit StaffPanel to show phone privacy

abtyx committed:
change major bug in PrivacyCommand

abtyx committed:
edit equals method for Address and Email to include checking for privacy

abtyx committed:
make private information show up as private on staff panel

abtyx committed:
edit tests to accomodate  for changes

abtyx committed:
add junit testing for coveralls
change staffPanel assert to check for privacy

abtyx committed:
fix checkstyle

NgeowXiuQi committed:
Merge pull request #56 from abtyx/v1.3PrivacyInUI

V1.2 + 1.3 privacy in ui

valensia0711 committed:
Merge branch 'master' into wip/favourite

Commit:
fix conflict resolve problem

Commit:
fix conflict resolve problem

Commit:
fix tests

Commit:
AddressBookParserTest: add test for favourite command

IamRENCE committed:
Merge pull request #5 from CS2103-AY1819S1-T13-2/master

Merge from main 1.2/1.3

IamRENCE committed:
RateCommandParser: Add check if rating entered fulfils regex

IamRENCE committed:
RateCommandParser: Check for negative index

IamRENCE committed:
FeedbackCommandParser: Add check if feedback fulfils feedback regex

Commit:
fix for checkstyle

Commit:
fix for checkstyle

Commit:
add test for unfavourite

IamRENCE committed:
Add RateCommandSystemTest

IamRENCE committed:
Add tests for RateCommandSystemTest

IamRENCE committed:
RateCommandSystemTest: Fix test comment

IamRENCE committed:
Add FeedbackCommandSystemTest

IamRENCE committed:
Merge pull request #54 from CS2103-AY1819S1-T13-2/wip/favourite

add favourite command

IamRENCE committed:
RateCommandParser: Fix comment issue

IamRENCE committed:
FeedbackCommandParser: Fix tests to properly check for the correct parse failure

IamRENCE committed:
RateCommandParser: Fix tests to properly check for the correct parse failure

IamRENCE committed:
RateCommandParser: Coveralls check

IamRENCE committed:
Fix Coveralls Check

valensia0711 committed:
Merge pull request #58 from IamRENCE/master

Fix bug for Rating and Feedback Commands

abtyx committed:
update userdoc for privacy command

NgeowXiuQi committed:
Merge pull request #62 from abtyx/v1.3UserDocAsh

update userdoc for privacy command

NgeowXiuQi committed:
Update XMLAdaptedPerson.java : Commented out unnecessary checks

NgeowXiuQi committed:
Updated XmlAdaptedPerson.java : Removed unnecessary checks
Updated Salary.java : Commented out unnecessary code

NgeowXiuQi committed:
Update Salary.java : Removed unnecessary function

DanKhoo committed:
Merge pull request #63 from NgeowXiuQi/Coveralls_Fix

Update XMLAdaptedPerson.java : Commented out unnecessary checks

Commit:
user guide: add favourite command, remove lock-unlock command

Commit:
wip: add unfavourite command

Commit:
user guide: add coming soon to lock-unlock feature

DanKhoo committed:
Merge pull request #64 from CS2103-AY1819S1-T13-2/documentation

user guide: add favourite command, remove lock-unlock command

DanKhoo committed:
Update UserGuide.adoc

Commit:
add unfavourite command

valensia0711 committed:
Merge pull request #65 from DanKhoo/master

Update UserGuide.adoc

Commit:
update user guide and dev guide to include unfavourite command

Commit:
removed unused method for coveralls

NgeowXiuQi committed:
Merge pull request #67 from CS2103-AY1819S1-T13-2/documentation

update user guide and dev guide to include unfavourite command

Commit:
fix for coveralls

DanKhoo committed:
Merge pull request #66 from CS2103-AY1819S1-T13-2/wip/unfavourite

add unfavourite command

IamRENCE committed:
Merge pull request #6 from CS2103-AY1819S1-T13-2/master

Merge 1.3

IamRENCE committed:
Feedback: Add check for Profanity

IamRENCE committed:
FeedbackTest: Add tests to check if profanities get rejected

IamRENCE committed:
FeedbackCommandParser & RateCommandParser: Update minor bug where wrong error message is displayed

IamRENCE committed:
FeedbackCommandParserTest & RateCommandParserTest: Update tests to follow correct error message output

IamRENCE committed:
RateCommandSystemTest: Update test to display correct error message

IamRENCE committed:
FeedbackCommandSystemTest: Add test to check if profanity in input is rejected

IamRENCE committed:
README: Update Acknowledgements for Profanity Filter code

IamRENCE committed:
Revert "README: Update Acknowledgements for Profanity Filter code"

This reverts commit 6f7fee3ac6f19776c1240dd0a6fc4dee62d3ff49.

IamRENCE committed:
Update UserGuide on feedback and rate command

IamRENCE committed:
Create Bad_Words_List

IamRENCE committed:
Update Bad_Words_List

IamRENCE committed:
Delete dummy Bad_Words_List

IamRENCE committed:
Update input file to be txt instead of csv

IamRENCE committed:
Feedback: Improve code quality

IamRENCE committed:
Improve code coverage

IamRENCE committed:
UserGuide: Fix typos

DanKhoo committed:
Merge pull request #75 from IamRENCE/master

Profanity Filter for Feedback Command

Commit:
add UI for favourite command

Commit:
edit UI for favourite command

Commit:
fix for checkstyle

Commit:
add sort command

Commit:
fix bug when sorting with favourited persons

Commit:
add tests for sort command

NgeowXiuQi committed:
Update UserGuide : Added Net Salary note in Edit Command

NgeowXiuQi committed:
Edited UserGuide : Removed formula

abtyx committed:
update userdoc for privacy

IamRENCE committed:
Update Feedback profanity Filter to work for jar file

IamRENCE committed:
Feedback: Fix checkstyle

IamRENCE committed:
Test fix NoClassDefFoundError in travis

NgeowXiuQi committed:
Merge pull request #79 from abtyx/v1.3UserDocAsh

update userdoc for privacy

NgeowXiuQi committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T13-2/main into UserGuide_Update

NgeowXiuQi committed:
Update UserGuide : Shifted Salary, OT Hours, OT Rate and Pay Deductibles to have its own section in the user guide

abtyx committed:
Merge pull request #78 from NgeowXiuQi/UserGuide_Update

Update UserGuide : Added Net Salary note in Edit Command

IamRENCE committed:
Create dummy file

IamRENCE committed:
Add Bad_Words_List.txt

IamRENCE committed:
Delete dummy file

IamRENCE committed:
Feedback: Edit bad words text source

IamRENCE committed:
Merge branch 'master' of https://github.com/IamRENCE/main

IamRENCE committed:
Delete wrong Bad_Words_List.txt

IamRENCE committed:
Create Dummy File

IamRENCE committed:
Add Bad_Words_List.txt

IamRENCE committed:
Delete Dummy File

IamRENCE committed:
Update Bad_Words_List.txt

DanKhoo committed:
Merge pull request #76 from CS2103-AY1819S1-T13-2/wip/favourite

add UI for favourite command

DanKhoo committed:
Merge pull request #80 from IamRENCE/master

Profanity filter to work for jar file

Commit:
fix for checkstyle

Commit:
fix tests

Commit:
fix tests

Commit:
resolve issue: undo and redo did not work properly in regards to favourite/unfavourite

IamRENCE committed:
Update UserGuide.adoc

Remove deleteall command in user guide

IamRENCE committed:
Update UserGuide.adoc

Add information that `rate`, `feedback` and `privacy` commands are undoable.

IamRENCE committed:
Merge pull request #100 from CS2103-AY1819S1-T13-2/resolve_issue

resolve issue: undo and redo for favourite/unfavourite

DanKhoo committed:
Phone.java: Update VALIDATION_REGEX

DanKhoo committed:
Update UserGuide.adoc

DanKhoo committed:
Update all phone numbers according to the validation regex

DanKhoo committed:
Merge pull request #101 from CS2103-AY1819S1-T13-2/IamRENCE-patch-1

Update UserGuide.adoc

IamRENCE committed:
Merge pull request #7 from CS2103-AY1819S1-T13-2/master

Merge since Mock PE

IamRENCE committed:
Test Validation Phone Regex

IamRENCE committed:
Merge pull request #103 from DanKhoo/master

Update Phone validation regex

IamRENCE committed:
Merge pull request #8 from CS2103-AY1819S1-T13-2/master

Merge update phone validation regex

IamRENCE committed:
Merge branch 'DanKhoo-master'

DanKhoo committed:
Add HideStaffPanelEvent.java for Undo/Redo Command

DanKhoo committed:
DeleteCommand.java: Add event trigger when list is empty

DanKhoo committed:
MainWindow.java: Add handleHideStaffPanelEvent method

DanKhoo committed:
Add event trigger when edit, feedback, privacy, rate, redo, undo commands on staffPanel

DanKhoo committed:
Merge pull request #102 from CS2103-AY1819S1-T13-2/update-userguide

Update UserGuide.adoc

DanKhoo committed:
Update VALIDATION_REGEX & message to only allow alphabetical characters

DanKhoo committed:
Update relevant tests according to the validation regex for Name and Manager

DanKhoo committed:
StaffPanel.fxml:Wrap label text for name, department and email

IamRENCE committed:
Merge pull request #104 from DanKhoo/fix_UI_bug

Fix ui bug

IamRENCE committed:
Merge pull request #81 from CS2103-AY1819S1-T13-2/wip/sort

add sort command

DanKhoo committed:
Add find features

DanKhoo committed:
Implement find feature for department + tests

DanKhoo committed:
Implement find feature for manager name + tests

DanKhoo committed:
Implement find feature for email + tests

DanKhoo committed:
Implement find feature for email + tests

DanKhoo committed:
update

DanKhoo committed:
AddressBookParser.java: Add find commands

DanKhoo committed:
FindCommand: update find command word to find-n

DanKhoo committed:
Update typicalPerson.java & .xml

IamRENCE committed:
Merge pull request #106 from DanKhoo/findCommands

Find commands

IamRENCE committed:
Merge pull request #9 from CS2103-AY1819S1-T13-2/master

Merge from Week 12

IamRENCE committed:
Replace "Address Book" elements to "SSENISUB" in documentations and "Ssenisub" to code (for checkstyle compliance)

IamRENCE committed:
Fox checkstyle issues

IamRENCE committed:
Fix missed checkstyle

IamRENCE committed:
Fix more checkstyle with regards to import ordering and variable naming

IamRENCE committed:
SortCommandTest: Fix checkstyle

DanKhoo committed:
Merge pull request #107 from IamRENCE/master

Replace "Address Book" elements to "SSENISUB"

DanKhoo committed:
StaffPanel.fxml:Set vgrow and wrap text

DanKhoo committed:
Set character limit for each attributes

DanKhoo committed:
Update Ui.png

DanKhoo committed:
Update Ui.png

DanKhoo committed:
Fix validation regex

IamRENCE committed:
UserPrefs: Rename data output file to "ssenisub.xml"

IamRENCE committed:
Add Sequence Diagrams

IamRENCE committed:
Update DeveloperGuide with Feedback command features

IamRENCE committed:
Update UserGuide with Feedback features

IamRENCE committed:
Merge pull request #108 from DanKhoo/fix_ui_bug

Fix ui bug

valensia0711 committed:
Merge pull request #109 from IamRENCE/master

Update documentation for Feedback command

IamRENCE committed:
SampleDataUtil: Make 1st person favourited

IamRENCE committed:
Merge pull request #10 from CS2103-AY1819S1-T13-2/master

Merge UI changes

IamRENCE committed:
Merge branch 'master' of https://github.com/IamRENCE/main

IamRENCE committed:
SampleDataUtil: Undo "favourite 1st person by default"

Commit:
add sortByDept, sortByRatingUp, sortByRatingDown commands

IamRENCE committed:
Add PPP for IamRENCE

IamRENCE committed:
Update IamRENCE PPP

DanKhoo committed:
Merge pull request #111 from IamRENCE/master

Add PPP for IamRENCE

IamRENCE committed:
Merge pull request #11 from CS2103-AY1819S1-T13-2/master

Merge pull request #111 from IamRENCE/master

IamRENCE committed:
Fix tag typo in UserGuide

DanKhoo committed:
Update UserGuide.adoc

DanKhoo committed:
Merge pull request #112 from IamRENCE/master

Fix tag typo in UserGuide

abtyx committed:
add command is now unable to add a person with existing phone or email
edit command is now unable to edit into a existing phone or email

IamRENCE committed:
Merge pull request #12 from CS2103-AY1819S1-T13-2/master

Merge pull request #112 from IamRENCE/master

IamRENCE committed:
Update DeveloperGuide

IamRENCE committed:
Merge pull request #113 from DanKhoo/updateDocs

Update UserGuide.adoc

DanKhoo committed:
Merge pull request #115 from IamRENCE/master

Update DeveloperGuide

abtyx committed:
fix tests

abtyx committed:
PPP created

abtyx committed:
fix checkstyle

abtyx committed:
Update Ssenisub.java

abtyx committed:
Update EditCommand.java

DanKhoo committed:
Merge pull request #114 from abtyx/v1.4finalAsh

fix duplicate phone/email problem and edit into same person crash error

abtyx committed:
update userguide, devguide and PPP

abtyx committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T13-2/main into PPP

# Conflicts:
#	docs/UserGuide.adoc

DanKhoo committed:
Merge pull request #117 from abtyx/PPP

user doc, dev guide, PPP

DanKhoo committed:
Update all diagrams for Dev Guide

DanKhoo committed:
Update DeveloperGuide

DanKhoo committed:
Update UserGuide.adoc

DanKhoo committed:
Merge pull request #118 from DanKhoo/updateDocs

Update docs

Commit:
add tests for the different sort commands

Commit:
add valensia0711.adoc

Commit:
add parser tests for new commands to increase coverage

NgeowXiuQi committed:
Merge pull request #119 from CS2103-AY1819S1-T13-2/wip/sort_withparam

add sortDeptCommand, sortRatingDownCommand, sortRatingUpCommand

Commit:
minor changes in User Guide

NgeowXiuQi committed:
Added NgeowXiuQi.adoc : For PPP

Commit:
minor changes to user guide: grammar fix

valensia0711 committed:
Merge pull request #120 from NgeowXiuQi/master

Added NgeowXiuQi.adoc : For PPP

NgeowXiuQi committed:
Edited AboutUs.adoc to link to team members' PPP

NgeowXiuQi committed:
Updated my PPP (NgeowXiuQi.adoc)

NgeowXiuQi committed:
Corrected Spelling error in Ashburn's PPP

valensia0711 committed:
Merge pull request #121 from NgeowXiuQi/NgeowXiuQi_PPP

Edited AboutUs.adoc to link to team members' PPP and Updated my PPP

Commit:
fix typos and grammar errors

Commit:
reorder command list in user guide

Commit:
add sort commands to user guide

Commit:
fix typo

Commit:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T13-2/main into ppp

Commit:
fix for checkstyle: add newline at EOF

Commit:
add PPP for valensia0711

Commit:
fix: sortRatingDown should be sortRatingUp

NgeowXiuQi committed:
Merge pull request #123 from CS2103-AY1819S1-T13-2/documentation

minor edits to User Guide

NgeowXiuQi committed:
Merge pull request #124 from CS2103-AY1819S1-T13-2/ppp

add PPP for valensia0711

DanKhoo committed:
Add dankhoo.adoc ppp and AboutUs.adoc

abtyx committed:
Merge pull request #125 from DanKhoo/updateDocs

Add dankhoo.adoc ppp and AboutUs.adoc

abtyx committed:
add Activity Diagram to dev guide
update PPP

NgeowXiuQi committed:
Edited ngeowxiuqi.adoc to include changes

IamRENCE committed:
Merge pull request #126 from abtyx/PPP

add Activity Diagram to dev guide

NgeowXiuQi committed:
Rename NgeowXiuQi.adoc to ngeowxiuqi.adoc

DanKhoo committed:
Update asciidoctor.css

NgeowXiuQi committed:
Merge pull request #128 from CS2103-AY1819S1-T13-2/NgeowXiuQi-patch-1

Rename NgeowXiuQi.adoc to ngeowxiuqi.adoc

NgeowXiuQi committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T13-2/main into NgeowXiuQi_PPP

NgeowXiuQi committed:
Removed ngeowxiuqi.adoc PPP

DanKhoo committed:
Merge pull request #129 from CS2103-AY1819S1-T13-2/updateCSS

Update asciidoctor.css

DanKhoo committed:
Merge pull request #130 from NgeowXiuQi/NgeowXiuQi_PPP

Removed ngeowxiuqi PPP

NgeowXiuQi committed:
Re-added ngeowxiuqi.adoc

DanKhoo committed:
Merge pull request #132 from NgeowXiuQi/NgeowXiuQi_PPP

Re-added ngeowxiuqi.adoc

NgeowXiuQi committed:
Edited AboutUs to accomodate for my PPP

DanKhoo committed:
DeveloperGuide.adoc: Add manual test cases for add, select and find-n

valensia0711 committed:
Merge pull request #133 from NgeowXiuQi/NgeowXiuQi_PPP

Edited AboutUs to accomodate for my PPP

NgeowXiuQi committed:
Edited DeveloperGuide and my PPP

NgeowXiuQi committed:
Merge pull request #134 from DanKhoo/updateDocs

DeveloperGuide.adoc: Add manual test cases for add, select and find-n

DanKhoo committed:
Update dankhoo.adoc PPP

DanKhoo committed:
Merge pull request #135 from NgeowXiuQi/NgeowXiuQi_PPP

Edited DeveloperGuide and my PPP

IamRENCE committed:
Merge pull request #13 from CS2103-AY1819S1-T13-2/master

More updates

IamRENCE committed:
Merge pull request #136 from DanKhoo/updateDocs

Update dankhoo.adoc PPP

IamRENCE committed:
Merge pull request #14 from CS2103-AY1819S1-T13-2/master

Even more updates

IamRENCE committed:
Update Rate Command Sequence Diagrams

IamRENCE committed:
Update Feedback Command Sequence Diagram

IamRENCE committed:
Add Feedback Profanity Filter Sequence Diagram

IamRENCE committed:
Update DeveloperGuide with Profanity Filter Sequence Diagram

DanKhoo committed:
Merge pull request #137 from IamRENCE/master

Add Feedback Profanity Sequence Diagram

Commit:
edit developer guide to include sort commands and favourite/unfavourite sequence diagrams

Commit:
fix typo in developer guide for feedback command

Commit:
reorder contents in command implementation

Commit:
reorder contents in use cases

Commit:
reorder contents in manual testing guide

Commit:
add manual test for favourite command

Commit:
add manual test for unfavourite command

DanKhoo committed:
Merge pull request #138 from CS2103-AY1819S1-T13-2/documentation

minor edits and additions to developer guide

Commit:
edit PPP: valensia0711.adoc

IamRENCE committed:
Merge pull request #15 from CS2103-AY1819S1-T13-2/master

Documentation updates

IamRENCE committed:
Fix typo in DeveloperGuide

IamRENCE committed:
Fix typo in PPP

IamRENCE committed:
Merge pull request #139 from CS2103-AY1819S1-T13-2/documentation

edit PPP: valensia0711.adoc

DanKhoo committed:
Merge pull request #140 from IamRENCE/master

Fix typo in DeverloperGuide and PPP

DanKhoo committed:
Update README.adoc

NgeowXiuQi committed:
Merge pull request #141 from CS2103-AY1819S1-T13-2/updateDocs

Update README.adoc

DanKhoo committed:
Update AboutUs.adoc
